src/Tasks/system.design/stronglytypedresourcebuilder.cs

```diff
@@ -83,53 +85,42 @@ internal sealed class ResourceData
             internal ResourceData(Type type, String valueAsString)
             {
                 Type = type;
+                TypeAssemblyQualifiedName = type.AssemblyQualifiedName;
+                TypeFullName = type.FullName;
                 ValueAsString = valueAsString;
             }
 
+            internal ResourceData(string typeName, string typeAssemblyQualifiedName)
```

Comment:
When reading the code that calls the constructors, it's confusing because the second argument means something entirely different depending on whether the first argument is a `Type` or just a `string` type name.  I'm not sure there's a great way to make this clearer though.